processes/record_frbrizer.py

```diff
@@ -0,0 +1,120 @@
+from collections import defaultdict
+from lxml import etree
+import re
+
+from logger import create_log
+from mappings.oclc_bib import OCLCBibMapping
+from mappings.oclcCatalog import CatalogMapping
+from managers import DBManager, OCLCCatalogManager, RedisManager
+from model import Record
+from .record_buffer import RecordBuffer
+
+
+logger = create_log(__name__)
+
+
+class RecordFRBRizer:
+
+    def __init__(self, db_manager: DBManager):
+        self.db_manager = db_manager
+
+        self.oclc_catalog_manager = OCLCCatalogManager()
+        
+        self.redis_manager = RedisManager()
+        self.redis_manager.createRedisClient()
+
+        self.record_buffer = RecordBuffer(db_manager=self.db_manager)
+
+    def frbrize_record(self, record: Record) -> Record:
+        owi_to_oclc_numbers = self._add_works_for_record(record=record)
+        self._add_edition_records(owi_to_oclc_numbers=owi_to_oclc_numbers)
+
+        self.record_buffer.flush()
+
+        record.frbr_status = 'complete'
+
+        self.db_manager.session.add(record)
+        self.db_manager.session.commit()
+
+        logger.info(f'FRBRized record: {record}')
+
+        return record
+
+    def _add_works_for_record(self, record: Record) -> defaultdict:
+        queryable_ids = self._get_queryable_identifiers(record.identifiers)
+        author = record.authors[0].split('|')[0] if record.authors else None
+        title = record.title
+
+        owi_to_oclc_numbers = defaultdict(set)
+
+        for id, id_type in queryable_ids:
+            if self.redis_manager.checkSetRedis('classify', id, id_type):
+                continue
+
+            search_query = self.oclc_catalog_manager.generate_search_query(identifier=id, identifier_type=id_type)
+            owi_to_oclc_numbers.update(self._add_works(self.oclc_catalog_manager.query_bibs(query=search_query)))
+
+        if len(owi_to_oclc_numbers) == 0 and author and title:
+            search_query = self.oclc_catalog_manager.generate_search_query(author=author, title=title)
+            owi_to_oclc_numbers.update(self._add_works(self.oclc_catalog_manager.query_bibs(query=search_query)))
+
+        return owi_to_oclc_numbers
+
+    def _add_works(self, oclc_bibs: list) -> set:
```

Comment:
Seems like this typehint might be off? I think we're returning a dict here (or really, looks like `dict[set[str]]`?)

Reply:
oops!